faker/providers/date_of_birth/__init__.py

```diff
@@ -0,0 +1,48 @@
+# coding=utf-8
+
+from __future__ import unicode_literals
+
+from .. import BaseProvider
+
+
+localized = True
+
+
+class Provider(BaseProvider):
+
+    # The most common Age of Majority in the world is 18.
+    #
+    # https://en.wikipedia.org/wiki/Age_of_majority
+    #
+    # As published in Nature magazine, it's suggested that that while
+    # there will still be occasional outliers, there is a natural limit
+    # to human lifespan of about 115 years old. This seems like a
+    # reasonable max value for our purposes.
+    # 
+    # https://www.nature.com/news/human-age-limit-claim-sparks-debate-1.20750
+
+    age_range = {}
+    age_range['minor'] = {'minimum': 0, 'maximum': 18}
+    age_range['adult'] = {'minimum': 18, 'maximum': 115}
+
+    def date_of_birth(self, stage_of_life='adult'):
```

Comment:
Instead of `stage_of_life`, I'd rather use numbers, like a `min_age` and a `max_age` parameters.